src/wovnio/wovnphp/API.php

```diff
@@ -19,17 +21,21 @@ public static function url($store, $headers, $original_content, $request_options
         ksort($store->settings);
         $settings_hash = md5(serialize($store->settings));
         $cache_key_string = "(token=$token&settings_hash=$settings_hash&body_hash=$body_hash&path=$path&lang=$lang)";
+
         if ($request_options->getCacheDisableMode() || $request_options->getDebugMode()) {
```

Comment:
Just an idea, but how about move `$headers->isSearchEngineBot()` to here like the following.
```
if ($request_options->getCacheDisableMode() || $request_options->getDebugMode() || $headers->isSearchEngineBot()) {
```

Then, we can remove `getSearchEngineBotTimeStamp` and make source code simple.

Reply:
Hmm. I separated them because the two timestamps are different. Not only are they two different representations, (one is unix epoch type, another is ISO 8601), but they have different underlining "time". The timestamp for debug mode and cache disable mode is updated for every new request, but the timestamp for search engine is updated every 20 minutes.